azurerm/internal/services/privatedns/resource_arm_private_dns_txt_record.go

```diff
@@ -0,0 +1,237 @@
+package privatedns
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPrivateDnsTxtRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPrivateDnsTxtRecordCreateUpdate,
+		Read:   resourceArmPrivateDnsTxtRecordRead,
+		Update: resourceArmPrivateDnsTxtRecordCreateUpdate,
+		Delete: resourceArmPrivateDnsTxtRecordDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				// lower-cased due to the broken API https://github.com/Azure/azure-rest-api-specs/issues/6641
+				ValidateFunc: validate.LowerCasedString,
+			},
+
+			// TODO: make this case sensitive once the API's fixed https://github.com/Azure/azure-rest-api-specs/issues/6641
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"zone_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"record": {
```

Comment:
Do you think any other attributes will be place under record? If not, we could probably shorten this to just be a TypeSet with string elements
```
Type:     schema.TypeSet,
					Required: true,
					Elem: &schema.Schema{
						Type: schema.TypeString,
					},
```

Reply:
Thanks, I wasn't sure about this actually as I wanted to try and keep it similar to the [azurerm_dns_txt_record](https://github.com/terraform-providers/terraform-provider-azurerm/blob/c916aa6db4142add1760a9b39bc668eb8b1dfcde/azurerm/internal/services/dns/resource_arm_dns_txt_record.go#L54-L66) resource for consistency. Do you think it's worth changing both resources (and deprecating the `value` key in the existing resource)?

Reply:
Ahhh, yea. I get that. I think that's just leftover from trying to match the terraform schema to the api struct but it's not a best practice. I don't think we can match it at this time because we couldn't keep the same name. We could just note it as a todo for 3.0